tensorflow/python/training/optimizer.py

```diff
@@ -385,7 +385,7 @@ def _assert_valid_dtypes(self, tensors):
   def _valid_dtypes(self):
     """Valid types for loss, variables and gradients.
 
-    Subclasses should override to allow other types.
+    Subclasses should override to allow other float types.
```

Comment:
Just to confirm, are float types the only types that could potentially added here? Or are doubles, ints, etc also possible?